test/tracecontext/sample_rate.js

```diff
@@ -0,0 +1,218 @@
+    agent._conf.transactionSampleRate = 0.499
+    // const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+    // })
+    const transaction = startSampledTransaction()
+    t.equals(transaction.sample_rate, 0.499, 'sample rate set')
+
+    t.end()
+  })
+
+  test.test('prefers tracestate value', function (t) {
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+      childOf: TRACEPARENT_RECORDED,
+      tracestate: 'es=s:0.7654321'
+    })
+    t.equals(transaction.sample_rate, 0.7654321, 'sample rate set')
+    t.end()
+  })
+
+  // test that a sampled/recorded transaction produces spans and transactions
+  // with a sample_rate property equal to the sample rate
+  test.test('recorded transactions have sample rate', function (t) {
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+      childOf: TRACEPARENT_RECORDED,
+      tracestate: 'es=s:0.7654321'
+    })
+    const span = transaction.startSpan('foo')
+    t.equals(transaction.sample_rate, 0.7654321, 'sample rate set')
+    t.equals(span.sample_rate, 0.7654321, 'sample rate set')
+    t.end()
+  })
+
+  // test that a unsampled/unrecorded transaction produces a transaction
+  // with a sample_rate property of 0, and produces no spans
+  test.test('unrecorded transactions have a sample rate of 0', function (t) {
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+      childOf: TRACEPARENT_NOTRECORDED,
+      tracestate: 'es=s:0'
+    })
+    const span = transaction.startSpan('foo')
+    t.equals(transaction.sample_rate, 0, 'sample rate set')
+    t.ok(!span, 'span not started')
+    t.end()
+  })
+
+  // test that a sampled/recorded root transaction produces samples and
+  // transactions with a sample_rate equal to the sample rate
+  test.test('recorded root transactions have a sample rate of 0', function (t) {
+    agent._conf.transactionSampleRate = 1
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing')
+    const span = transaction.startSpan('foo')
+    t.equals(transaction.sample_rate, 1, 'sample rate set')
+    t.equals(span.sample_rate, 1, 'sample rate set')
+    t.end()
+  })
+
+  // test that an unsampled/unrecorded root  transaction produces a transaction
+  // with a sample_rate property of 0, and produces no spans
+  test.test('recorded root transactions have a sample rate of 0', function (t) {
+    agent._conf.transactionSampleRate = 0
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing')
+    const span = transaction.startSpan('foo')
+    t.equals(transaction.sample_rate, 0, 'sample rate set')
+    t.ok(!span, 'no span for unsampled transactions')
+    t.end()
+  })
+
+  // test that an unsampled transaction from a non-zero sample rate
+  // still produces a final serialized span with a sample rate of 0
+  test.test('recorded root transactions have a sample rate of 0', function (t) {
```

Comment:
copypasta: Update this test name to match what is tested.

Reply:
✅